src/extract.rs

```diff
@@ -13,6 +14,27 @@ pub(crate) enum FileType {
     Plaintext,
 }
 
+impl Default for FileType {
+    fn default() -> Self {
+        Self::Plaintext
+    }
+}
+
+impl<P: AsRef<Path>> From<P> for FileType {
+    /// Detect if the given path points to a Markdown, HTML, or plaintext file.
+    fn from(p: P) -> FileType {
+        let path = p.as_ref();
+        match path.extension() {
+            Some(ext) => match ext.to_str().expect("Couldn't convert OsStr to str") {
```

Comment:
I wonder if we want to use `TryFrom` instead of `From` for this function to be able to return that error here instead of panicking.

Reply:
I can't really imagine a realistic use case where `OsStr` would not be convertible to `str` (UTF-8). It would require some malice from the users (such as including zero bytes or whatever nastiness). Because of that, I'm inclined to say we should not pollute the API and just stick with `expect`.

Alternatively, we could convert each of the literals below to `OsString` and match against that... :thinking: Using `lazy_static` should actually make it cheaper too...

Let me see what I can do :wink:

Reply:
Actually, ended up using `PartialEq` for `OsString` https://doc.rust-lang.org/std/ffi/struct.OsString.html#impl-PartialEq%3C%26%27_%20str%3E